pkg/cluster/control_plane_endpoint.go

```diff
@@ -0,0 +1,56 @@
+package cluster
+
+import (
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+	"github.com/giantswarm/azure-admission-controller/pkg/key"
+)
+
+func validateClusterNetwork(cluster capiv1alpha3.Cluster, baseDomain string) error {
+	if cluster.Spec.ClusterNetwork == nil {
+		return microerror.Maskf(errors.InvalidOperationError, "ClusterNetwork can't be null")
+	}
+
+	if *cluster.Spec.ClusterNetwork.APIServerPort != key.ControlPlaneEndpointPort {
+		return microerror.Maskf(errors.InvalidOperationError, "ClusterNetwork.APIServerPort can only be set to %d", key.ControlPlaneEndpointPort)
+	}
+
+	if cluster.Spec.ClusterNetwork.ServiceDomain != key.ServiceDomain(cluster.Name, baseDomain) {
+		return microerror.Maskf(errors.InvalidOperationError, "ClusterNetwork.ServiceDomain can only be set to %s", key.ServiceDomain(cluster.Name, baseDomain))
+	}
+
+	if cluster.Spec.ClusterNetwork.Services == nil {
+		return microerror.Maskf(errors.InvalidOperationError, "ClusterNetwork.Services can't be null")
+	}
+
+	if !reflect.DeepEqual(cluster.Spec.ClusterNetwork.Services.CIDRBlocks, []string{key.ClusterNetworkServiceCIDR}) {
```

Comment:
Will be the length always 1? I'm asking because order matters, but I believe it's fine because it will be only one.

Reply:
it is for now